packages/node_modules/@webex/internal-plugin-devices/src/devices.js

```diff
@@ -319,8 +334,86 @@ const Devices = WebexPlugin.extend({
 
   // Helper method members
 
-  canRegister() {}, // Validates if the device can register
-  checkNetworkReachability() {}, // Checks for internal network reachability.
+  /**
+   * Determine if registration methods can be performed. This method utilizes
+   * the `services` plugin to confirm if the appropriate service urls are
+   * available for device registration.
+   *
+   * @param {boolean} [wait=false] - Willing to wait on registration.
+   * @returns {Promise<void, Error>}
+   */
+  canRegister(wait = false) {
+    this.logger.info('devices: validating if registration can occur');
+
+    // Destructure the services plugin for ease of reference.
+    const {services} = this.webex.internal;
+
+    // Validate if the service url exists in the service catalog.
+    if (services.get('wdm')) {
+      return Promise.resolve();
+    }
+
+    // If wait is true, wait for the catalog to populate.
+    if (wait) {
+      return services.waitForCatalog('postauth',
```

Comment:
Should we do one extra check after this returns to make sure the wdm catalog was populated by postauth?

Reply:
The `postauth` catalog will always contain wdm. I can add the additional validation just in case if you believe it's necessary.

Reply:
I think it will be worthwhile just in case